src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -495,6 +442,67 @@ public static bool Initialize(string userSpecifiedPath)
             }
         }
 
+
+#if NET45 || NET46 || NET47
+        private static bool LoadNetFrameworkDelegates(string userSpecifiedPath)
+        {
+            string path = userSpecifiedPath;
+            if (path == null)
+            {
+                // in net45, librdkafka.dll is not in the process directory, we have to load it manually
+                // and also search in the same folder for its dependencies (LOAD_WITH_ALTERED_SEARCH_PATH)
+                var is64 = IntPtr.Size == 8;
+                var baseUri = new Uri(Assembly.GetExecutingAssembly().GetName().EscapedCodeBase);
+                var baseDirectory = Path.GetDirectoryName(baseUri.LocalPath);
+                var dllDirectory = Path.Combine(
```

Comment:
Maybe move these four path blocks to a for loop and try to load each one until one succeeds?

Reply:
that's a good idea, but minor detail + this is copy/paste from existing implementation with lots of runtime, and any change opens up possibility of bugs + it takes effort.